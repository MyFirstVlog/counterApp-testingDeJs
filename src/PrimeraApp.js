// dos componentes, por funciones o clases,m por practicas se debe hacer por funciones
import React from 'react';
import { Fragment } from 'react';
import PropTypes from 'prop-types'

//Se espera que regrese algo html 

const PrimeraApp = ({saludo, subtitulo}) => {
    //  Solo puedo devolver un tag a la vez o fragmento  a la vez, JSX Fragment <> </>, si pongo () despues del retun solo me deja devolver uno 
    // return ( <h1> oliwi</h1>)

    // const saludo = 'Hola Mundo'

 
    // if(!saludo) throw new Error('el saludo es necesario')
    console.log(saludo)

    //No imprime booleanos

    // Nom imprime objetos asi : (toca hacerles stringify)
    /**
     * const saludo = {
        nombre: 'Alejo',
        edad : 92
    }

    3 es ele espacio 
     */
    return (
        <>
            <h1>{saludo}!!!</h1>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{subtitulo}</p>
        </>
    )
}

//Propiedades y valores por defecto y obligatorios

PrimeraApp.propTypes = {
    saludo : PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un Subtitulo'
}

export default PrimeraApp

/***
 * Dos qequivalencias  al ahora de devolver fragmentos
 * 
 * Manera poco eficientes por creacion de divs sin uso adicional 
 * return ( 
        <div>
            <h1>Hola Mundo</h1>
            <p>Mi Primera App con React Js</p>
        </div>
    )

    Estos dos no crean divs

    return( <>
            <h1>Hola Mundo</h1>
            <p>Mi Primera App con React Js</p>
        </div>
    </> )

    Esta mete una importacion adicional
    return (
        <Fragment>
            <h1>Hola Mundo</h1>
            <p>Mi Primera App con React Js</p>
        </Fragment>
    )
 */