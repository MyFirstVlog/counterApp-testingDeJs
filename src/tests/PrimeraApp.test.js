import React from 'react';
import { render } from "@testing-library/react"
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp"


describe('Pruebas en componente primera app', () => {
    
    // test('Debe mostar el mensaje hola soy naruto', () => {
        
    //     const saludo = 'Hola Soy Naruto'

    //     const {getByText} = render(<PrimeraApp saludo= {saludo} />) //producto renderizado 

    //     //Usar enzyme

    //     //expect(getByText(saludo)).toBeInTheDocument() //Verificar si esta en el root
    
    // })

    test('debe mostar <PrimeraApp/> correctamente', () => {

        const saludo = 'Hola Soy Naruto'

        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot() //Toma foto de prueba

    })

    test('debe mostar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola Soy Naruto'
        const subtitulo = 'Soy el sub puesto a fza bruta'

        const wrapper = shallow(
                    <PrimeraApp 
                        saludo={saludo}
                        subtitulo={subtitulo}
                    />
                    )
        const textoParrafo = wrapper.find('p').text()
        
        expect(textoParrafo).toBe(subtitulo)

    })
    
    
    
})
