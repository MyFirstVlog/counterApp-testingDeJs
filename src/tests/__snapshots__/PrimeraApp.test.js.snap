// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente primera app debe mostar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Soy Naruto
    !!!
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
